### ISSUE TITLE
AI competitions dont produce useful models  Luke Oakden-Rayner

### ISSUE THREAD

**Issue by guevara** [guevara/read-it-later#5500]
AI competitions don&rsquo;t produce useful models &ndash; Luke Oakden-Rayner<br>
<br>
https://ift.tt/34XG8nO<br>
<br>
About 0 Minutes<br>
<br>
<div>
<div>
<p>A huge new CT brain dataset was released today, with the goal of training models to detect intracranial haemorrhage. So far, it looks pretty good, although I haven&rsquo;t dug into it in detail yet (and the devil is often in the detail).</p>
<p>The <a href="https://www.rsna.org/en/education/ai-resources-and-training/ai-image-challenge">dataset has been released for a competition,</a> which obviously lead to the usual friendly rivalry on Twitter:</p>
<p><img alt="ai competition1" src="https://lukeoakdenrayner.files.wordpress.com/2019/09/ai-competition1.png?w=429&amp;h=493"></p>
<p>Of course, this lead to cynicism from the usual suspects as well.</p>
<p><img alt="ai competition2" src="https://lukeoakdenrayner.files.wordpress.com/2019/09/ai-competition2.png?w=444&amp;h=420"></p>
<p>And the conversation continued from there, with thoughts ranging from &ldquo;but since there is a hold out test set, how can you overfit?&rdquo; to &ldquo;the proposed solutions are never intended to be applied directly&rdquo; (the latter from a previous competition winner).</p>
<p>As the discussion progressed, I realised that while we &ldquo;all know&rdquo; that competition results are more than a bit dubious in a clinical sense, I&rsquo;ve never really seen a compelling explanation for why this is so.</p>
<p>Hopefully that is what this post is, an explanation for why competitions are not really about building useful AI systems.</p>
<hr>
<p>&nbsp;</p>
<h3><strong>Let&rsquo;s have a battle</strong></h3>
<p><img alt="748284__safe_animated_screencap_equestriagirls_rainbowrocks_spoiler-colon-rainbowrocks_sonatadusk_adagiodazzle_ariablaze_thedazzlings" src="https://lukeoakdenrayner.files.wordpress.com/2019/09/748284__safe_animated_screencap_equestriagirls_rainbowrocks_spoiler-colon-rainbowrocks_sonatadusk_adagiodazzle_ariablaze_thedazzlings.gif?w=656"></p>
<h5>Nothing wrong with a little competition.*</h5>
<p>So what is a competition in medical AI? Here are a few options:</p>
<ul>
<li>getting teams to try to solve a clinical problem</li>
<li>getting teams to explore how problems might be solved and to try novel solutions</li>
<li>getting teams to build a model that performs the best on the competition test set</li>
<li>a waste of time</li>
</ul>
<p>Now, I&rsquo;m not so jaded that I jump to the last option (what is valuable to spend time on is a matter of opinion, and clinical utility is only one consideration. More on this at the end of the article).</p>
<p>But what about the first three options? Do these models work for the clinical task, and do they lead to broadly applicable solutions and novelty, or are they only good in the competition and not in the real world?</p>
<p>(Spoiler: I&rsquo;m going to argue the latter).</p>
<hr>
<p>&nbsp;</p>
<h3><strong>Good models and bad models</strong></h3>
<p>Should we expect this competition to produce good models? Let&rsquo;s see what one of the organisers says.</p>
<p><img alt="ai competition3" src="https://lukeoakdenrayner.files.wordpress.com/2019/09/ai-competition3.png?w=379&amp;h=457"></p>
<p>Cool. Totally agree. The lack of large, well-labeled datasets is the biggest major barrier to building useful clinical AI, so this dataset should help.</p>
<p>But saying that the dataset can be useful is not the same thing as saying the competition will produce good models.</p>
<p>So to define our terms, let&rsquo;s say that a <strong>good model&nbsp;</strong>is a model that can detect brain haemorrhages on <i>unseen&nbsp;</i>data (cases that the model has no knowledge of).</p>
<p>So conversely, a <strong>bad model</strong> is&nbsp;one that doesn&rsquo;t detect brain haemorrhages in unseen&nbsp;data.</p>
<p>These definitions will be non-controversial. Machine Learning 101. I&rsquo;m sure the contest organisers agree with these definitions, and would prefer their participants to be producing good models rather than bad models. In fact, they have clearly set up the competition in a way designed to promote good models.</p>
<p>It just isn&rsquo;t enough.</p>
<hr>
<p>&nbsp;</p>
<h3><strong>Epi vs ML, FIGHT!</strong></h3>
<p><img alt="source" src="https://lukeoakdenrayner.files.wordpress.com/2019/09/source.gif?w=656"></p>
<h5>If only academic arguments were this cute</h5>
<p>ML101 (now personified) tells us that the way to control overfitting is to use a hold-out test set, which is data that has not been seen during model training. This simulates seeing new patients in a clinical setting.</p>
<p>ML101 also says that hold-out data is only good for one test. If you test multiple models, then even if you don&rsquo;t cheat and leak test information into your development process, your best result is probably an outlier which was only better than your worst result by chance.</p>
<p>So competition organisers these days produce hold-out test sets, and only let each team run their model on the data once. Problem solved, says ML101. The winner only tested once, so there is no reason to think they are an outlier, they just have the best model.</p>
<p>Not so fast, buddy.</p>
<p>Let me introduce you to Epidemiology 101, who<span>&nbsp;claims to have</span><span>&nbsp;a magic coin.&nbsp;</span></p>
<p><img alt="fsnuw6sujx28bh5itvip" src="https://lukeoakdenrayner.files.wordpress.com/2019/09/fsnuw6sujx28bh5itvip.gif?w=300&amp;h=135"></p>
<p><span>Epi101 tells you to flip the coin 10 times. If you get 8 or more heads, that confirms the coin is magic (while the assertion is clearly nonsense, you play along since you know that 8/10 heads equates to a p-value of &lt;0.05 for a fair coin, so it must be legit).</span></p>
<p><img alt="tumblr_pkeicpjcfb1r65eeio5_r1_500" src="https://lukeoakdenrayner.files.wordpress.com/2019/09/tumblr_pkeicpjcfb1r65eeio5_r1_500.gif?w=656"></p>
<p>Unbeknownst to you, Epi101 does the same thing with 99 other people, all of whom think they are the only one testing the coin. What do you expect to happen?</p>
<p>If the coin is totally normal and not magic, around 5 people will find that the coin is special. Seems obvious, but think about this in the context of the individuals. Those 5 people all only ran a single test. According to them, they have statistically significant evidence they are holding a &ldquo;magic&rdquo; coin.</p>
<p>Now imagine you aren&rsquo;t flipping coins. Imagine you are all running a model on a competition test set. Instead of wondering if your coin is magic, you instead are hoping that your model is the best one, about to earn you $25,000.</p>
<p>Of course, you can&rsquo;t submit more than one model. That would be cheating. One of the models could perform well, the equivalent of getting 8 heads with a fair coin, just by chance.</p>
<p>Good thing there is a rule against it submitting multiple models, or any one of the other 99 participants and their 99 models could win, just by being lucky&hellip;</p>
<p><img alt="tumblr_mvn6uwbyep1shi48to5_250" src="https://lukeoakdenrayner.files.wordpress.com/2019/09/tumblr_mvn6uwbyep1shi48to5_250.gif?w=656"></p>
<hr>
<p>&nbsp;</p>
<h3><strong>Multiple hypothesis testing</strong></h3>
<p>The effect we saw with Epi101&rsquo;s coin applies to our competition, of course. Due to random chance, some percentage of models will outperform other ones,&nbsp;<em>even if they are all just as good as each other</em>. Maths doesn&rsquo;t care if it was one team that tested 100 models, or 100 teams.</p>
<p><span>Even if certain models</span><span>&nbsp;</span><em>are</em><span>&nbsp;</span><span>better than others in a meaningful sense^, unless you truly believe that the winner is uniquely able to ML-wizard, you have to accept that at least some other participants would have achieved similar results, and thus the winner only won because they got lucky. The real &ldquo;best performance&rdquo; will be somewhere back in the pack, probably above average but below the winner^^.</span></p>
<p><img alt="ai competition4" src="https://lukeoakdenrayner.files.wordpress.com/2019/09/ai-competition4-1.png?w=620&amp;h=311"></p>
<p>Epi101 says this effect is called multiple hypothesis testing. In the case of a competition, you have a ton of hypotheses &ndash; that each participant was better than all others. For 100 participants, 100 hypotheses.</p>
<p>One of those hypotheses, taken in isolation, might show us there is a winner with statistical significance (p&lt;0.05). But taken together, even if the winner has a calculated &ldquo;winning&rdquo; p-value of less than 0.05, that doesn&rsquo;t mean we only have a 5% chance of making an unjustified decision. In fact, if this was coin flips (which is easier to calculate but not absurdly different), we would have a greater than 99% chance that one or more people would &ldquo;win&rdquo; and come up with 8 heads!</p>
<p>That is what an AI competition winner is; an individual who happens to get 8 heads while flipping fair coins.</p>
<p>Interestingly, while ML101 is very clear that running 100 models yourself and picking the best one will result in overfitting, they rarely discuss this &ldquo;overfitting of the crowds&rdquo;. Strange, when you consider that almost all ML research is done of heavily over-tested public datasets &hellip;</p>
<p>So how do we deal with multiple hypothesis testing? It all comes down to the cause of the problem, which is the data. Epi101 tells us that any test set is a biased version of the target population. In this case, the target population is &ldquo;all patients with CT head imaging, with and without intracranial haemorrhage&rdquo;. Let&rsquo;s look at how this kind of bias might play out, with a toy example of a small hypothetical population:</p>
<p><img alt="brain bleeds 1" src="https://lukeoakdenrayner.files.wordpress.com/2019/09/brain-bleeds-1.png?w=675&amp;h=140"></p>
<p>In this population, we have a pretty reasonable &ldquo;clinical&rdquo; mix of cases. 3 intra-cerebral bleeds (likely related to high blood pressure or stroke), and two traumatic bleeds (a subdural on the right, and an extradural second from the left).</p>
<p>Now let&rsquo;s sample this population to build our test set:</p>
<p><img alt="brain bleeds 2" src="https://lukeoakdenrayner.files.wordpress.com/2019/09/brain-bleeds-2.png?w=388&amp;h=135"></p>
<p>Randomly, we end up with mostly extra-axial (outside of the brain itself) bleeds. A model that performs well on this test will not necessarily work as well on real patients. In fact, you might expect a model that is really good at extra-axial bleeds at the expense of intra-cerebral bleeds to win.</p>
<p>But Epi101 doesn&rsquo;t only point out problems. Epi101 has&nbsp;a solution.</p>
<hr>
<p>&nbsp;</p>
<h3><strong>So powerful</strong></h3>
<p>There is only one way to have an unbiased test set &ndash; if it includes the entire population! Then whatever model does well in the test will also be the best in practice, because you tested it on all possible future patients (which seems difficult).</p>
<p>This leads to a very simple idea &ndash; your test results become more reliable as the test set gets larger. We can actually predict how reliable test sets are using power calculations.</p>
<p><img alt="image4" src="https://lukeoakdenrayner.files.wordpress.com/2019/09/image4.png?w=451&amp;h=257"></p>
<p>These are power curves. If you have a rough idea of how much better your &ldquo;winning&rdquo; model will be than the next best model, you can estimate how many test cases you need to reliably show that it is better.</p>
<p>So to find out if you model is 10% better than a competitor, you would need about 300 test cases. You can also see how exponentially the number of cases needed grows as the difference between models gets narrower.</p>
<p>Let&rsquo;s put this into practice. If we look at another medical AI competition, the <a href="https://www.kaggle.com/c/siim-acr-pneumothorax-segmentation/overview">ACR pneumothorax segmentation challenge</a>, we see that the difference in Dice scores (ranging between 0 and 1) is negligible at the top of the leaderboard. Keep in mind that this competition had a dataset of 3200 cases (and that is being generous, they don&rsquo;t all contribute to the Dice score equally).</p>
<p><img alt="pneumo challenge" src="https://lukeoakdenrayner.files.wordpress.com/2019/09/pneumo-challenge.png?w=663&amp;h=521"></p>
<p>So the difference between the top two was 0.0014 &hellip; let&rsquo;s chuck that into a sample size calculator.</p>
<p><img alt="sample size 1" src="https://lukeoakdenrayner.files.wordpress.com/2019/09/sample-size-1-1.png?w=498&amp;h=330"></p>
<p>Ok, so to show a significant difference between these two results, you would need 920,000 cases.</p>
<p>But why stop there? We haven&rsquo;t even discussed multiple hypothesis testing yet. This absurd number of cases needed is simply if there was ever only one hypothesis, meaning only two participants.</p>
<p>If we look at the leaderboard, there were 351 teams who made submissions. The rules say they could submit two models, so we might as well assume there were at least 500 tests. This has to produce some outliers, just like 500 people flipping a fair coin.</p>
<p>Epi101 to the rescue. Multiple hypothesis testing is really common in medicine, particularly in &ldquo;big data&rdquo; fields like genomics. We have spent the last few decades learning how to deal with this. The simplest reliable way to manage this problem is called the Bonferroni correction^^.</p>
<p>The Bonferroni correction is super simple: you divide the p-value by the number of tests to find a &ldquo;statistical significance threshold&rdquo; that has been adjusted for all those extra coin flips. So in this case, we do 0.05/500. Our new p-value target is 0.0001, any result worse than this will be considered to support the null hypothesis (that the competitors performed equally well on the test set). So let&rsquo;s plug that in our power calculator.</p>
<p><img alt="sample size 2" src="https://lukeoakdenrayner.files.wordpress.com/2019/09/sample-size-2-1.png?w=474&amp;h=306"></p>
<p>Cool! It only increased a bit&hellip; to 2.6 million cases needed for a valid result :p</p>
<p>Now, you might say I am being very unfair here, and that there must be some small group of good models at the top of the leaderboard that are not clearly different from each other^^^. Fine, lets be generous. Surely no-one will complain if I compare the 1st place model to the 150th model?</p>
<p><img alt="sample size 3" src="https://lukeoakdenrayner.files.wordpress.com/2019/09/sample-size-3-1.png?w=480&amp;h=317"></p>
<p>&nbsp;</p>
<p>So still more data than we had. In fact, I have to go down to the <em>192nd</em> placeholder to find a result where the sample size was enough to produce a &ldquo;statistically significant&rdquo; difference.</p>
<p>But maybe this is specific to the pneumothorax challenge? What about other competitions?</p>
<p>In <a href="https://stanfordmlgroup.github.io/competitions/mura/">MURA</a>, we have a test set of 207 x-rays, with 70 teams submitting &ldquo;no more than two models per month&rdquo;, so lets be generous and say 100 models were submitted. Running the numbers, the &ldquo;first place&rdquo; model is only significant versus the 56th placeholder and below.</p>
<p>In the <a href="https://www.kaggle.com/c/rsna-pneumonia-detection-challenge/overview">RSNA Pneumonia Detection Challenge</a>, there were 3000 test images with 350 teams submitting one model each. The first place was only significant compared to the 30th place and below.</p>
<p>And to really put the cat amongst the pigeons, what about outside of medicine?</p>
<p><img alt="imagenet" src="https://lukeoakdenrayner.files.wordpress.com/2019/09/imagenet-1.png?w=586&amp;h=318"></p>
<p>As we go left to right in ImageNet results, the improvement year on year slows (the effect size decreases) and the number of people who have tested on the dataset increases. I can&rsquo;t really estimate the numbers, but knowing what we know about multiple testing does anyone really believe the SOTA rush in the mid 2010s was anything but crowdsourced overfitting?</p>
<hr>
<p>&nbsp;</p>
<h3><strong>So what are competitions for?</strong></h3>
<p><img alt="b26" src="https://lukeoakdenrayner.files.wordpress.com/2019/09/b26.gif?w=481&amp;h=263"></p>
<p>They obviously aren&rsquo;t to reliably find the best model. They don&rsquo;t even really reveal useful techniques to build great models, because we don&rsquo;t know which of the hundred plus models actually used a good, reliable method, and which method just happened to fit the under-powered test set.</p>
<p>You talk to competition organisers &hellip; and they mostly say that competitions are for publicity. And that is enough, I guess.</p>
<p>AI competitions are fun, community building, talent scouting, brand promoting, and attention grabbing.</p>
<p>But AI competitions are&nbsp;<strong>not</strong> to develop useful models.</p>
<p>&nbsp;</p>
<hr>
<h5>* I have a young daughter, don&rsquo;t judge me for my encyclopaedic knowledge of My Little Pony.**</h5>
<h5>** not that there is anything wrong with My Little Pony***. Friendship&nbsp;<em>is</em> magic. There is just an unsavoury internet element that matches my demographic who is <em>really</em> into the show. I&rsquo;m no brony.</h5>
<h5>*** barring the near complete white-washing of a children&rsquo;s show about multi-coloured horses.</h5>
<h5>^ we can actually understand model performance with our coin analogy. Improving the model would be equivalent to bending the coin. If you are good at coin bending, doing this will make it more likely to land on heads, but unless it is 100% likely you still have no guarantee to &ldquo;win&rdquo;. If you have a 60%-chance-of-heads coin, and everyone else has a 50% coin, you objectively have the best coin, but your chance of getting 8 heads out of 10 flips is still only 17%. Better than the 5% the rest of the field have, but remember that there are 99 of them. They have a cumulative chance of over 99% that <em>one</em> of them will get 8 or more heads.</h5>
<h5>^^ people often say the Bonferroni correction is a bit conservative, but remember, we are coming in skeptical that these models are actually different from each other. We&nbsp;<em>should&nbsp;</em>be conservative.</h5>
<h5>^^^ do please note, the top model here got $30,000 and the second model got nothing. The competition organisers felt that the distinction was reasonable.</h5>
</div>
</div>
<br>
<br>
<br>
<br>
<br>
<br>
via Luke Oakden-Rayner https://ift.tt/2qjzenH<br>
<br>
September 20, 2019 at 04:52PM
